src/components/forms/profile/update-user-profile-form.tsx

```diff
@@ -34,36 +34,96 @@ const UpdateUserProfileForm: React.FC<{ userProfileData?: Profile }> = ({
       <Form
         formID={formID}
         onSubmit={async () => {
-          setFormID(uuidv4());
-          return;
+          await fetchJson("/api/profile/user-profile", "PATCH", {
+            userProfilePayload: userProfile,
+          }).then(async (response) => {
+            const parsedResponse = await response.json();
+
+            if ("updatedUserProfile" in parsedResponse) {
+              router && router.push("/account/profile");
+              return;
+            }
+
+            throw new Error("Something went wrong");
+          });
```

Comment:
Answering to this comment https://github.com/fewlinesco/connect-account/pull/541#discussion_r626550345
I think we can do:

```suggestion
          await fetchJson("/api/profile/user-profile", "PATCH", {
            userProfilePayload: userProfile,
          }).then(async (response) => {
            const parsedResponse = await response.json();
            router && router.push("/account/profile");
            return;
          }).catch(() => throw new Error("Something went wrong");)
```
> Not sure that works though, so maybe more:

```suggestion
          await fetchJson("/api/profile/user-profile", "PATCH", {
            userProfilePayload: userProfile,
          }).then(async (response) => {
            if (response.status !== HttpStatus.OK) {
              throw new Error("Something went wrong");
             }
            const parsedResponse = await response.json();
            router && router.push("/account/profile");
            return;
            }
          });
```

Because the payload can be full of empty fields but should not a problem since the API page will return a "not" 200 code if it's the case. Am I right?

Reply:
Or a mix of the two 🤔 
I mean, If something went wrong on the API side, it will not return the data we want, in any case. So the api should sent a not 200 status code and then, if it's the case, the front should throw.

Reply:
In case of an error, the parsed response will have a `message` key, coming from the web-error, but no `updatedUserProfile `. See https://github.com/fewlinesco/connect-account/blob/main/src/components/forms/update-identity-form.tsx for example.

The first one would not work. The second one would work though.

Also, I can't see any cases for now that we will do anything with the data returned by Connect.Profile in this flow, as we redirect. I'd prefere keeping the same implementation across our code base.

If we wanna change this, we'll have to do it everywhere :).

Wdyt?

Reply:
but...
> Not sure if I should do this now, as it would make this tedious

Yeah, as you prefer 🙂

Reply:
> Wdyt?

I agree :)